terraform/gcp/main.tf

```diff
@@ -0,0 +1,134 @@
+locals {
+  activate_apis = [
+    "artifactregistry.googleapis.com",
+    "run.googleapis.com",
+    "secretmanager.googleapis.com"
+  ]
+}
+
+data "google_project" "project" {}
+
+resource "google_project_service" "project_services" {
+  for_each                   = toset(local.activate_apis)
+  project                    = var.gcp_project
+  service                    = each.value
+  disable_on_destroy         = true
+  disable_dependent_services = true
+}
+
+resource "google_storage_bucket" "buz_schemas" {
+  name          = "buz-schemas-${data.google_project.project.number}"
+  location      = "US"
+  force_destroy = true
+}
+
+resource "google_pubsub_topic" "valid_topic" {
+  name = "buz-valid"
+}
+
+resource "google_pubsub_topic" "invalid_topic" {
+  name = "buz-invalid"
+}
+
+resource "google_secret_manager_secret" "buz_config" {
+  secret_id = "buz-config"
+
+  replication {
+    user_managed {
+      replicas {
+        location = var.gcp_region
+      }
+    }
+  }
+
+  depends_on = [
+    google_project_service.project_services
+  ]
+}
+
+resource "google_secret_manager_secret_version" "buz_config" {
+  secret      = google_secret_manager_secret.buz_config.id
+  secret_data = file("gcp.yml")
+}
+
+resource "google_artifact_registry_repository" "buz_repository" {
+  location      = var.gcp_region
+  repository_id = "buz-repository"
+  format        = "DOCKER"
+
+  depends_on = [
+    google_project_service.project_services
+  ]
+}
+
+resource "google_project_iam_binding" "buz_config_secret_access" {
+  project = var.gcp_project
+  role    = "roles/secretmanager.secretAccessor"
+  members = [
+    "serviceAccount:${data.google_project.project.number}-compute@developer.gserviceaccount.com"
+  ]
+
+  # condition {
```

Comment:
As far as I can tell, this should be correct but wasn't working until I removed the condition. I can try adding it back if everything else looks good.

Reply:
👀